Updated Travis to include newer Python versions

catleeball opened the pull request.

Commit:
Updated Travis to include newer Python versions

houglum commented:
This seems reasonable since we need to test Python 3.6-3.7.

Per the discussion offline, can we require the higher version, i.e. `>=0.9.6`, of HTTPretty in requirements.txt in this PR as well?  That should allow us to move 3.7 out of the allowed_failures quarantine box :)

All Python versions passed in my fork's tests (well, everything except 2.6, which was already failing) when I bumped to the new version of HTTPretty:
https://travis-ci.org/houglum/boto/builds/508647285?utm_source=github_status&utm_medium=notification

houglum committed:
Update HTTPretty version to something current

The previous commit that changed the HTTPretty version, https://github.com/boto/boto/commit/cf7ede6acc5109b894a2251ced753984cffff991, mentioned that 0.8.7 caused tests to hang, but it provided no bug link, logs, etc.  So, after seeing passing TravisCI tests when using v0.9.6, I can only assume that this new version of HTTPretty fixes whatever bugs were occurring when that previous commit was added.

catleeball committed:
Merge pull request #3 from houglum/patch-1

Update HTTPretty version to something current

houglum commented:
Also worth noting: We're bumping HTTPretty's version because that fixes test errors in Python 3.7 (i.e. https://github.com/gabrielfalcao/HTTPretty/issues/337), which looked something like this:

```
======================================================================
ERROR: Check connection re-use
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/boto/boto/tests/unit/test_connection.py", line 337, in test_multi_commands
    "POST")
  File "/home/travis/build/boto/boto/boto/connection.py", line 1119, in make_request
    return self._mexe(http_request)
  File "/home/travis/build/boto/boto/boto/connection.py", line 946, in _mexe
    request.body, request.headers)
  File "/opt/python/3.7.1/lib/python3.7/http/client.py", line 1229, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/opt/python/3.7.1/lib/python3.7/http/client.py", line 1275, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/opt/python/3.7.1/lib/python3.7/http/client.py", line 1224, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/opt/python/3.7.1/lib/python3.7/http/client.py", line 1016, in _send_output
    self.send(msg)
  File "/opt/python/3.7.1/lib/python3.7/http/client.py", line 956, in send
    self.connect()
  File "/home/travis/build/boto/boto/boto/https_connection.py", line 137, in connect
    self.sock = context.wrap_socket(sock, server_hostname=self.host)
  File "/opt/python/3.7.1/lib/python3.7/ssl.py", line 412, in wrap_socket
    session=session
  File "/opt/python/3.7.1/lib/python3.7/ssl.py", line 817, in _create
    super(SSLSocket, self).__init__(**kwargs)
TypeError: super(type, obj): obj must be an instance or subtype of type
```

houglum commented:
LGTM.  @mfschwartz, could you merge this when you get a chance?  This increases the test matrix to cover Python 3.6 and 3.7.  As noted in the previous PR, the Py2.6 environment was already failing.
